src/commands/autocomplete/values.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+
+import {flags} from 'cli-engine-command'
+import AutocompleteBase from '.'
+import Plugins from '../../plugins'
+import ACCache from '../../cache'
+import path from 'path'
+
+export default class AutocompleteValues extends AutocompleteBase {
+  static topic = 'autocomplete'
+  static command = 'values'
+  static description = 'generates autocomplete values'
+  static hidden = true
+  static flags = {
+    // don't require cmd or flag
+    // we want it to run silently
+    // or autocomplete will use any
+    // flag errors as options
+    cmd: flags.string({description: '', char: 'c'}),
+    flag: flags.string({description: '', char: 'f'})
+  }
+
+  async run () {
+    try {
+      this.errorIfWindows()
+
+      if (!this.flags.cmd) this.out.error('Missing required value for --cmd')
+      if (!this.flags.flag) this.out.error('Missing required value for --flag')
+
+      const plugins = new Plugins(this.out)
+      await plugins.load()
+      let Command = await plugins.findCommand(this.flags.cmd)
+      if (!Command) this.out.error(`Command ${this.flags.cmd} not found`)
+      let long = this.flags.flag.replace(/^-+/, '')
+      let flags = Command ? Command.flags : {}
+      let flag = flags[long]
+      if (!flag) this.out.error(`Flag ${long} not found`)
+      if (flag.completion && flag.completion.options) {
+        let flagCache = path.join(this.config.cacheDir, 'completions', long)
```

Comment:
@RasPhilCo do we also want to include the command in the path so that we do not collide on name in case it is defined differently in two commands? (not today)